packages/components/src/components/askem/Askem.ts

```diff
@@ -0,0 +1,61 @@
+import type { AskemConfigs } from './types';
+
+class Askem {
+  constructor(config: AskemConfigs) {
+    if (!config.apiKey) {
+      throw new Error('Api key is required.');
+    }
+
+    this.initialize(config);
+  }
+
+  private initialize({
+    apiKey,
+    scriptUrl = 'https://cdn.reactandshare.com/plugin/rns.js',
+    disabled = false,
+  }: AskemConfigs) {
+    if (disabled) {
+      return;
+    }
+
+    if (typeof window === 'undefined') {
+      return;
+    }
+
+    window.rnsData = window.rnsData || {};
+    window.rnsData.apiKey;
+    if (!window.rnsData.apiKey) {
+      window.rnsData.apiKey = apiKey;
+    }
+
+    const doc = document;
+    const scriptElement = doc.createElement('script');
+    const scripts = doc.getElementsByTagName('script')[0];
+
+    scriptElement.type = 'text/javascript';
+    scriptElement.async = true;
+    scriptElement.defer = true;
+    scriptElement.src = scriptUrl;
```

Comment:
Cosmetic thing, but would an object.assign be better? 
```suggestion
Object.assign(scriptElement, {
	type: 'text/javascript',
	async: true,
	defer: true,
	src: scriptUrl
})
```